### PR TITLE
User can search for more relevant results.

### PR DESCRIPTION
**Pull request by aidanbdh**

### CODE REVIEW

File: search.js [hunk 1]
```diff
@@ -1,72 +1,56 @@
...
-          results.push(i);
-        };
-      };
+  text = fixText(text);
+  for (let i = 0; i < text.length; i++) {
+    for (let j = text[i].length; j > 0; j--) {
+      text[i] = text[i].slice(0, j);
+      const resultsA = profiles.filter(value => {
+        const name = fixText(value.name);
+        return (name.filter((value, index) => {return name[index].indexOf(text[i]) !== -1}))[0];
```
Comment (thebearingedge) [inline search.js]: Check out `Array.prototype.find`

File: search.js [hunk 2]
```diff
@@ -76,19 +60,18 @@ const searchBox = function() {
 
 const searchPage = function() {
   let views = [];
-  for (const i = 0; i < results.length; i++) {
-   const result = results[i];
-   const view = createDomElement('div',{'data-navigation': result},[
+  for (let i = 0; i < results.length; i++) {
```
Comment (thebearingedge) [inline search.js]: Any reason not to use `map` here?

File: search.js [hunk 3]
```diff
@@ -1,72 +1,56 @@
...
   let l = results.length;
   if (l > 3) { l = 3; };
   //Create results
-
   let views = [];
   const box = document.createElement('div');
   for (let i = 0; i < l; i++) {
-    const result = results[i];
-    const view = createDomElement('div',{'data-navigation': result},[
+    const view = createDomElement('div',{'data-navigation': profiles.indexOf(results[i])},[
```
Comment (thebearingedge) [inline search.js:44]: Any reason not to use `map` here?

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: search.js (hunk 3 above)
Comment (aidanbdh) [reply search.js:44]: I used the for loop to limit the number of results the dropdown displays to three

File: search.js (hunk 1 above)
Comment (aidanbdh) [reply search.js]: Thanks!

File: search.js (hunk 2 above)
Comment (aidanbdh) [reply search.js]: The deed is done. The for loop is no more.

**Pull request merged**